src/Microsoft.NetCore.Analyzers/Core/Security/SetHttpOnlyForHttpCookie.cs

```diff
@@ -0,0 +1,205 @@
+                return PropertySetAbstractValue.GetInstance(PropertySetAbstractValueKind.Flagged);
+            });
+
+        private static readonly PropertyMapperCollection PropertyMappers = new PropertyMapperCollection(
+            new PropertyMapper(
+                "HttpOnly",
+                (PointsToAbstractValue pointsToAbstractValue) =>
+                   PropertySetAbstractValueKind.Unflagged));
+
+        private static readonly HazardousUsageEvaluatorCollection HazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+                    new HazardousUsageEvaluator(
+                        HazardousUsageEvaluatorKind.Return,
+                        PropertySetCallbacks.HazardousIfAllFlaggedAndAtLeastOneKnown));
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    WellKnownTypeProvider wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemWebHttpCookie, out INamedTypeSymbol httpCookieSymbol))
+                    {
+                        return;
+                    }
+
+                    PooledHashSet<(IOperation Operation, ISymbol ContainingSymbol)> rootOperationsNeedingAnalysis = PooledHashSet<(IOperation, ISymbol)>.GetInstance();
+
+                    compilationStartAnalysisContext.RegisterOperationBlockStartAction(
+                        (OperationBlockStartAnalysisContext operationBlockStartAnalysisContext) =>
+                        {
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    ISimpleAssignmentOperation simpleAssignmentOperation =
+                                        (ISimpleAssignmentOperation)operationAnalysisContext.Operation;
+                                    if (simpleAssignmentOperation.Target is IPropertyReferenceOperation propertyReferenceOperation &&
+                                        httpCookieSymbol.Equals(propertyReferenceOperation.Property.ContainingType) &&
+                                        propertyReferenceOperation.Property.Name == "HttpOnly" &&
+                                        simpleAssignmentOperation.Value.ConstantValue.HasValue &&
+                                        simpleAssignmentOperation.Value.ConstantValue.Value.Equals(false))
```

Comment:
`simpleAssignmentOperation.Value.ConstantValue.Value` - can't this be null? Can we instead check `simpleAssignmentOperation.Value.ConstantValue.Value == false`?

Reply:
Yes, it's bool, can't be assigned with null.